app/models/changeset/pull_request.rb

```diff
@@ -38,9 +44,19 @@ def self.changeset_from_webhook(project, params = {})
     new(project.github_repo, data)
   end
 
+  # Webhook events that are valid should be related to a pr code push or someone adding [samson review]
+  # to the description. The actions related to a code push are 'opened' and 'synchronized'
+  # The 'edited' action gets sent when the PR description is edited. To trigger a deploy from an edit - it
+  # should only be when the edit is related to adding the text [samson review]
   def self.valid_webhook?(params)
     data = params['pull_request'] || {}
-    return false unless data['state'] == 'open'
+    action = params.dig('github', 'action')
+    return false unless data['state'] == 'open' && (VALID_ACTIONS.include? action)
+
+    if action == 'edited'
+      previous_desc = params['github']['changes']['body']['from']
```

Comment:
nitpick: you can also `dig` this value although it won't crash if one of the nested hashes is unexpectedly `nil` - depends on the behaviour you prefer